perun/profile/helpers.py

```diff
@@ -408,47 +414,32 @@ def is_key_aggregatable_by(profile: profiles.Profile, func: str, key: str, keyna
     return True
 
 
-@decorators.singleton
```

Comment:
This was actually kept as an optimization, mainly for test cases, since it is called a lot, and the recursive lookup of keys in the hierarchy is costly. Since you removed this function completely, it would be better to move the caching somewhere. However, I'm currently not keen on moving it to `safely_lookup_recursively`. I'd rather have this function `get_sort_order`, which would call the two lookups instead.